### PR TITLE
Gulp build system

### PR DESCRIPTION
**Pull request by dougrich**
Added the gulp build system we talked about offline. Re-organized folders around this. See the output in this file: [microluxe20_complete.pdf](https://github.com/kgrubb/microluxe20/files/74335/microluxe20_complete.pdf)

### CODE REVIEW

**Comment by dougrich** [comment]
Ugh, missed updating the filename in README.md. Will fix this.

File: src/markdown/microluxe20_complete.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+![title-img](/src/static/logo.png)
```
Comment (dougrich) [inline src/markdown/microluxe20_complete.md]: Not sure if you want me to do this for all files, but this is how you would put your logo at the beginning of the document.

**Comment by kgrubb** [comment]
Uh oh, my only problem with this so far is the new pdf conversion's tables seem to be broken. I'm betting it has to do with table heads in the markdown. I'll play with this some and see if we can get the tables sorted out. Aside from that, everything looks great!

**Comment by dougrich** [comment]
It's styled by the CSS, so you should be able to just add stylings for table and th however you want.

**Comment by kgrubb** [comment]
Yeah, I'm already messing with the css to clean up the pdf some. I'm going to try and shrink the pdf some more, since the new pdf's are approx 8-10mb and double the page-length (from 24-48). I'm liking this build method, though. It's a lot faster and easier to manage.

**Comment by dougrich** [comment]
Hmm. I'm not seeing any difference in file size, but the page count difference seems to be because of the font size and padding. Could you send one of the PDFs you created?

I also didn't append the OGL like in your docs.

**Comment by kgrubb** [comment]
Ok so I've found that whenever I build the docs on my mac, the pdf filesize is approx 6-10mb in size. However, on my linux and windows computers, the filesize is around 200kb. Strange...I'm not sure why that's the case.

As for the OGL licensing, I think its safe the way it is. The repo has a copy of the licensing, and I'll make sure to include the licensing as a separate document inside the downloadable releases.

**Comment by dougrich** [comment]
Okay did some digging.

Build process depends on gulp-markdown-pdf, which wraps markdown-pdf. Issue on them is [here](https://github.com/alanshaw/markdown-pdf/issues/37).

They then point to their underlying dependency of PhantomJS, and the issue on them is [here](
https://github.com/ariya/phantomjs/issues/10373).

So looks like it's an issue with text in OSX PDFs, where instead of text fields it produces curves. I use PhantomJS for a number of things, so I'll keep an eye on it and check it when it drops a new version.

**Comment by kgrubb** [comment]
Thanks for the heads up with that. I added a notice in the readme, so that people building on a mac don't get confused/upset.

On another note, I've played with the css some and got the tables looking better, and the docs are back to approx 25 pages. I staged a new release on the 31st. I also set up a super barebones github pages website for the project. I just wanted to say thanks for helping out!

**Pull request merged**